lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HDFSCommandRunner.cs

```diff
@@ -0,0 +1,194 @@
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Org.Apache.REEF.IO.FileSystem.Hadoop.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IO.FileSystem.Hadoop
+{
+    /// <summary>
+    /// Helper class to execute commands
+    /// </summary>
+    internal sealed class HdfsCommandRunner
+    {
+        /// <summary>
+        /// The name of the <code>hdfs</code> command.
+        /// </summary>
+        private const string HdfsCommandName = "hdfs.cmd";
+
+        /// <summary>
+        /// The folder within <code>HadoopHome</code> that contains the <code>hdfs</code> command.
+        /// </summary>
+        private const string BinFolderName = "bin";
+
+        private static readonly Logger Logger = Logger.GetLogger(typeof(HdfsCommandRunner));
+
+        /// <summary>
+        /// Path to hdfs.cmd
+        /// </summary>
+        private readonly string _hdfsCommandPath;
+
+        /// <summary>
+        /// The number of retries on HDFS commands.
+        /// </summary>
+        private readonly int _numberOfRetries;
+
+        /// <summary>
+        /// The timeout on each of the retries.
+        /// </summary>
+        private readonly int _timeOutInMilliSeconds;
+
+        [Inject]
+        private HdfsCommandRunner([Parameter(typeof(NumberOfRetries))] int numberOfRetries,
+            [Parameter(typeof(CommandTimeOut))] int timeOutInMilliSeconds,
+            [Parameter(typeof(HadoopHome))] string hadoopHome)
+        {
+            _numberOfRetries = numberOfRetries;
+            _timeOutInMilliSeconds = timeOutInMilliSeconds;
+
+            if (hadoopHome == HadoopHome.DefaultValue)
+            {
+                var hadoopHomeFromEnv = Environment.GetEnvironmentVariable("HADOOP_HOME");
+                if (null == hadoopHomeFromEnv)
+                {
+                    throw new Exception("HADOOP_HOME not set and no path to the hadoop installation provided.");
+                }
+                _hdfsCommandPath = GetFullPathToHdfsCommand(hadoopHomeFromEnv);
+            }
+            else
+            {
+                _hdfsCommandPath = GetFullPathToHdfsCommand(hadoopHome);
```

Comment:
Good idea. I added some code that tries the user-provided folder first. If that fails, it tries the environment variable.